redbot/cogs/mod/events.py

```diff
@@ -42,18 +42,19 @@ async def check_duplicates(self, message):
         return False
 
     async def check_mention_spam(self, message):
-        guild = message.guild
-        author = message.author
+        guild, author = message.guild, message.author
+        mention_spam = await self.config.guild(guild).mention_spam.all()
 
-        max_mentions = await self.config.guild(guild).ban_mention_spam()
-        if max_mentions:
-            mentions = set(message.mentions)
-            if len(mentions) >= max_mentions:
+        mentions = set(message.mentions)
+        if mention_spam["ban"]:
+            if len(mentions) >= mention_spam["ban"]:
                 try:
                     await guild.ban(author, reason=_("Mention spam (Autoban)"))
```

Comment:
`ban` defaults to deleting a day's worth of messages from a user (including the mention message). I know this was the previous behavior, but doing so creates ghost pings, so I don't know if we want to override that and not delete any messages.

Reply:
I'm good with whichever, I do think maybe keeping the message content there is better, so disallowing it from removing messages is good.